fdbserver/DDShardTracker.actor.cpp

```diff
@@ -1524,6 +1524,44 @@ struct DataDistributionTrackerImpl {
 				when(GetMetricsListRequest req = waitNext(self->getShardMetricsList)) {
 					self->actors.add(fetchShardMetricsList(self, req));
 				}
+				when(ServerTeamInfo req = waitNext(self->triggerStorageQueueRebalance)) {
+					TraceEvent e("TriggerDataMoveStorageQueueRebalance", self->distributorId);
+					e.detail("Server", req.serverId);
+					e.detail("Teams", req.teams.size());
+					int64_t maxShardWriteTraffic = 0;
+					KeyRange shardToMove;
+					ShardsAffectedByTeamFailure::Team selectedTeam;
+					for (const auto& team : req.teams) {
+						for (auto const& shard : self->shardsAffectedByTeamFailure->getShardsFor(team)) {
+							for (auto it : self->shards->intersectingRanges(shard)) {
+								if (it->value().stats->get().present()) {
+									int64_t shardWriteTraffic =
+									    it->value().stats->get().get().metrics.bytesWrittenPerKSecond;
+									if (shardWriteTraffic > maxShardWriteTraffic &&
+									    shardWriteTraffic > SERVER_KNOBS->REBALANCE_STORAGE_QUEUE_SHARD_PER_KSEC_MIN) {
```

Comment:
Is this condition `shardWriteTraffic > SERVER_KNOBS->REBALANCE_STORAGE_QUEUE_SHARD_PER_KSEC_MIN` used to avoid conflicts with `mergeShard`?

Will the following scenario possible?
There are many relatively large shards on a storage server (they are not eligible for merge due to size), while each of them has a relatively low traffic, but the accumulative traffic is causing a long storage server queue?

Reply:
> Is this condition `shardWriteTraffic > SERVER_KNOBS->REBALANCE_STORAGE_QUEUE_SHARD_PER_KSEC_MIN` used to avoid conflicts with `mergeShard`?
 
 Yes. Moving data is heavy. So we do not want to move a shard which is not able to effectively reduce the traffic to the server.

Reply:
> Will the following scenario possible? There are many relatively large shards on a storage server (they are not eligible for merge due to size), while each of them has a relatively low traffic, but the accumulative traffic is causing a long storage server queue?

Nice catch. I think we could remove this line